charts/eks/templates/rbac/clusterrole.yaml

```diff
@@ -18,7 +18,10 @@ rules:
     resources: ["nodes", "nodes/status"]
     verbs: ["get", "watch", "list"]
   - apiGroups: [""]
-    resources: ["pods", "nodes/proxy", "nodes/metrics", "nodes/stats"]
+    resources: [ "pods", 
+     {{- if .Values.isolation.enabled | and .Values.isolation.nodeProxyPermission.enabled }} "nodes/proxy",
+     {{- else if eq .Values.isolation.enabled false}} "nodes/proxy",
```

Comment:
This conditions seems same as previous condition, i think we should remove this `else if` condition.

Also instead of defining this same condition in multiple places, you can create a helper file, where you can define your logic and then use(include) it in multiple places.

Reply:
Added a new rule for better readability as suggested in the upstream PR